src/rpy_diffusive_flux.cc

```diff
@@ -0,0 +1,166 @@
+#include "flyft/spherical_mesh.h"
+#include "flyft/rpy_diffusive_flux.h"
+
+#include <exception>
+
+namespace flyft
+{
+   
+void RPYDiffusiveFlux::compute(std::shared_ptr<GrandPotential> grand, std::shared_ptr<State> state)
+    {
+    setup(grand,state);
+    state->syncFields();
+    
+    auto excess = grand->getExcessFunctional();
+    auto external = grand->getExternalPotential();
+    if (excess)
+        {
+        excess->compute(state,false);
+        }
+    if (external)
+        {
+        external->compute(state,false);
+        }
+    
+    
+    // compute fluxes on the left edge of the volumes (exclude the first point)
+    const auto mesh = state->getMesh()->local().get();
+    if (!dynamic_cast<const SphericalMesh*>(mesh))
+        {
+        throw std::invalid_argument("Spherical geometry required");
+        }
+
+    for (const auto &i : state->getTypes())
+        {
+        const double a_i = 0.5 * diameters_(i);
+        auto rho_i = state->getField(i)->const_view();
+        auto flux_i = fluxes_(i)->view();
+
+        // fill flux with zeros initially
+        std::fill(flux_i.begin(), flux_i.end(), 0.);
+
+        // RPY flux from all species
+        for (const auto &j : state->getTypes())
+            {
+            const double a_j = 0.5 * diameters_(j);
+            auto rho_j = state->getField(j)->const_view();
+            auto mu_ex_j = (excess) ? excess->getDerivative(j)->const_view() : Field::ConstantView();
+            auto V_j = (external) ? external->getDerivative(j)->const_view() : Field::ConstantView();
+            bool no_flux = false;
+            for (int idx = 0; idx < mesh->shape(); ++idx)
+                {
+                if (V_j)
+                    {
+                    bool Vidx_inf = std::isinf(V_j(idx));
+                    if (Vidx_inf)
+                        {
+                        if (V_j(idx) > 0 && rho_j(idx) > 0)
+                            {
+                            // ERROR: can't be V=+inf and have positive rho
+                            }
+                        else if(V_j(idx) < 0)
+                            {
+                            // ERROR: can't have V=-inf ever, it is a mass sink
+                            }
+                        }
+                    no_flux = (Vidx_inf || std::isinf(V_j(idx-1)));
+                    }
+                if (!no_flux)
+                    {
+                    const auto x = mesh->lower_bound(idx);
+                    if (x==0)
+                        {
+                        flux_i(idx) = 0;
+                        }
+                    else
```

Comment:
Same thing here, you can use continue and lose the else to avoid another level of indentation:
```suggestion
                    if (x == 0)
                        {
                        flux_i(idx) = 0.;
                        continue;
                        }
```
You should also probably add a comment line explaining why this is.